mobius/examples/crinkle_arc.rs

```diff
@@ -42,30 +42,49 @@ fn get_orthog_circle(circle: Circle, a: Complex, b: Complex) -> Circle {
     }
 }
 
-fn arc_lerp(circle: Circle, a: Complex, b: Complex, t: f64) -> Complex {
+fn lerp(a: f64, b: f64, t: f64) -> f64 {
+    (1.0 - t) * a + t * b
+}
+
+fn compute_sub_arc(
+    original_circle: Circle,
+    circle: Circle,
+    a: Complex,
+    b: Complex,
+    t: f64,
+) -> CircularArc {
     let angle_a = circle.get_angle(a).unwrap();
     let angle_b = circle.get_angle(b).unwrap();
+    let angle_mid1 = lerp(angle_a, angle_b, t);
+    let angle_mid2 = angle_mid1 + PI;
 
-    let lerp_angle = (1.0 - t) * angle_a + t * angle_b;
+    let mid1 = circle.get_point(angle_mid1);
+    let mid2 = circle.get_point(angle_mid2);
 
-    circle.get_point(lerp_angle)
+    if original_circle.point_inside(mid1) {
+        CircularArc::from_circle_and_points(circle, a, mid1, b).unwrap()
+    } else {
+        CircularArc::from_circle_and_points(circle, a, mid2, b).unwrap()
+    }
 }
 
-fn arc_fractal(arc: CircularArc) -> (Mobius, Mobius) {
+fn arc_fractal(arc: CircularArc, t: f64) -> (Mobius, Mobius) {
     let CircularArc { circle, angles } = arc;
     let ArcAngles(angle_a, angle_c) = angles;
-    let t = 0.5;
 
     let a = circle.get_point(angle_a);
-    let angle_b = t * (angle_a + angle_c);
+    let angle_b = lerp(angle_a, angle_c, t);
     let b = circle.get_point(angle_b);
     let c = circle.get_point(angle_c);
 
     let circle_ab = get_orthog_circle(circle, a, b);
     let circle_bc = get_orthog_circle(circle, b, c);
 
-    let d = arc_lerp(circle_ab, b, a, t);
-    let e = arc_lerp(circle_bc, c, b, t);
+    let arc_ab = compute_sub_arc(circle, circle_ab, a, b, t);
+    let arc_bc = compute_sub_arc(circle, circle_bc, b, c, t);
```

Comment:
While the calculations will be the same, conceptually, the operation is `compute_orthogonal_arc(arc, a, b, t)`

For now, have this method compute the _inside_ portion of the arc, and have it go in the same direction as the original arc. In the future, I may want to have flags for this.

Reply:
Though also, even without the flags, `complement()` and `reverse()` get you the rest of the way.